src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/ApproximateWayCheck.java

```diff
@@ -0,0 +1,216 @@
+package org.openstreetmap.atlas.checks.validation.linear.edges;
+
+import static java.lang.Math.pow;
+import static java.lang.Math.sqrt;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import org.openstreetmap.atlas.checks.atlas.predicates.TypePredicates;
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.Location;
+import org.openstreetmap.atlas.geography.Segment;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Distance;
+
+/**
+ * This check flags edges that deviate from the assumed curve of a road by at least
+ * {@value DEVIATION_MINIMUM_LENGTH_DEFAULT} meters.
+ *
+ * @author v-brjor
+ */
+public class ApproximateWayCheck extends BaseCheck<Long>
+{
+
+    private static final long serialVersionUID = 1L;
+    private static final String EDGE_DEVIATION_INSTRUCTION = "Way {0,number,#} is crude";
```

Comment:
The instruction could be more descriptive like "Way {0,number,#} is crude.  Add more nodes/rearrange current nodes".